Dockerfile

```diff
@@ -1,34 +1,6 @@
-# java-builder: Stage to build a custom JRE (with jlink)
-FROM eclipse-temurin:11@sha256:f4ab40c615ec2bd6a09992f6ceebeca796752e58ea778ccc6439087a728ba3b3 AS java-builder
-
-# create a custom, minimized JRE via jlink
-RUN jlink --add-modules \
-# include required modules
-java.base,java.desktop,java.instrument,java.management,java.naming,java.scripting,java.sql,java.xml,jdk.compiler,\
-# jdk.unsupported contains sun.misc.Unsafe which is required by certain dependencies
-jdk.unsupported,\
-# add additional cipher suites
-jdk.crypto.cryptoki,\
-# add ability to open ZIP/JAR files
-jdk.zipfs,\
-# OpenSearch needs some jdk modules
-jdk.httpserver,jdk.management,\
-# MQ Broker requires management agent
-jdk.management.agent,\
-# required for Spark/Hadoop
-java.security.jgss,jdk.security.auth,\
-# Elasticsearch 7+ crashes without Thai Segmentation support
-jdk.localedata --include-locales en,th \
-    --compress 2 --strip-debug --no-header-files --no-man-pages --output /usr/lib/jvm/java-11 && \
-  cp ${JAVA_HOME}/bin/javac /usr/lib/jvm/java-11/bin/javac && \
-  cp -r ${JAVA_HOME}/include /usr/lib/jvm/java-11/include && \
-  mv /usr/lib/jvm/java-11/lib/modules /usr/lib/jvm/java-11/lib/modules.bk; \
-  cp -r ${JAVA_HOME}/lib/* /usr/lib/jvm/java-11/lib/; \
-  mv /usr/lib/jvm/java-11/lib/modules.bk /usr/lib/jvm/java-11/lib/modules; \
-  rm -rf /usr/bin/java ${JAVA_HOME} && ln -s /usr/lib/jvm/java-11/bin/java /usr/bin/java
```

Comment:
This optimization saves ~55 MB (as you mentioned in the PR description), so this change would increase the image size by ~14%. Is there a chance we could implement this optimization in the installer? Or are there any other ways to optimize the size of the Java package?

Reply:
With a325dfc79, the installer uses [jlink](https://docs.oracle.com/javase/9/tools/jlink.htm) to optimise the JRE as before. The install footprint post-optimisation is as follows:

```
 8  197M
11   62M
17   66M
21   75M
```
Java 8 is not optimised because it does not include jlink, which was introduced in Java 9.